flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/dialect/JDBCDialects.java

```diff
@@ -403,5 +413,38 @@ public int minTimestampPrecision() {
 			);
 
 		}
+
+		@Override
+		public void setRow(ResultSet resultSet, RowType rowType, Row reuse) throws SQLException {
+			for (int pos = 0; pos < rowType.getFieldCount(); pos++) {
+				LogicalType type = rowType.getTypeAt(pos);
+				Object v = resultSet.getObject(pos + 1);
+
+				if (type instanceof SmallIntType) {
+					reuse.setField(pos, ((Integer) v).shortValue());
+				} else if (type instanceof ArrayType) {
```

Comment:
IIUC, currently, only Postgres supports `ArrayType`?

Reply:
I don't know yet. My goal is to make Postgres data types fully work, so I didn't touch other dbs. It should be easy for other dbs to follow if they support array.